lightning/src/ln/channelmonitor.rs

```diff
@@ -726,12 +727,255 @@ impl PartialEq for ChannelMonitor {
 	}
 }
 
-impl ChannelMonitor {
-	pub(super) fn new(funding_key: &SecretKey, revocation_base_key: &SecretKey, delayed_payment_base_key: &SecretKey, htlc_base_key: &SecretKey, payment_base_key: &SecretKey, shutdown_pubkey: &PublicKey, our_to_self_delay: u16, destination_script: Script, logger: Arc<Logger>) -> ChannelMonitor {
+impl<ChanSigner: ChannelKeys + Writeable> ChannelMonitor<ChanSigner> {
```

Comment:
Hmm, *moving* code hunks in the same commit where they are changed makes it super hard to review. If you want to move a code hunk, best to do it in its own commit clearly tagged move-only so that its easy to verify.

Reply:
I had no choice, because of the need for `+ Writeable`.  But you are right, I should have split that to its own commit.  I'll look into doing that.

Reply:
Also related is my comment above that ChanSigner would benefit from being dynamic.

Reply:
Hmm? I've never heard of Rust caring about ordering of loose hunks in a file, even if they rely on trait impls defined later in that file?

Reply:
Oh, oops, I missed what you did there. So I'm confused, why does ChannelMonitor need to be Writeable now?

Reply:
This section is now under `impl<ChanSigner: ChannelKeys + Writeable>`, whereas the previous section is now under `impl<ChanSigner: ChannelKeys>`.  I tried putting everything under `+ Writeable`, but the compiler gets unhappy.

Reply:
Oh, our comments crossed.  Only the `write*` methods need the `+ Writeable` on `ChanSigner` - so that they can serialize the `keys` field.

Reply:
Ahhh, ok, right, now things make more sense. In any case, can you split the commit (you probably want git reset HEAD^1 && git commit -p or so) so that you first just have two impl ChannelMonitor blocks with the write above (or maybe below where the read is is better?) and then have the second commit that makes changes.

Reply:
Will do.

Reply:
The split is done.  `git show --color-moved HEAD^`